### PR TITLE
refactor: Health Controller DAO

### LINKED ISSUE #656: Refactor HealthEndpoint to use DAO

### PR DESCRIPTION
**Pull request by jwicks31**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Resolves #656 

- Adds `HealthDAO` to `Health` Module.
- Moved all calls to `mongodb` to `HealthDAO`
- Updated `HealthController` to use `HealthDAO`

### CODE REVIEW

File: src/Health/HealthController.ts [hunk 1]
```diff
@@ -96,15 +69,7 @@ export class HealthController {
...
-      { name: 'blockchainInfo' },
-      {
-        $set: {
-          blockchainInfo,
-        },
-      },
-      { upsert: true }
-    )
+    await this.healthDAO.updateBlockchainInfo(blockchainInfo)
     return blockchainInfo
```
Comment (wzalazar) [inline src/Health/HealthController.ts]: the type of return is wrong...

Comment (jwicks31) [reply src/Health/HealthController.ts]: @wzalazar can you help with what's wrong?

File: src/Health/HealthDAO.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      { name: 'walletInfo' },
+      {
+        $set: {
+          walletInfo,
+        },
+      },
+      { upsert: true }
+    )
+
+  readonly updateNetworkInfo: updateNetworkInfo = async networkInfo =>
```
Comment (lautarodragan) [inline src/Health/HealthDAO.ts]: no need for async/await in this case, can just 

```ts
readonly updateNetworkInfo: updateNetworkInfo = networkInfo =>
    this.collection.updateOne(
      { name: 'networkInfo' },
      {
        $set: {
          networkInfo,
        },
      },
      { upsert: true }
    )
```

File: src/Health/HealthController.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import BitcoinCore = require('bitcoin-core')
 import { inject, injectable } from 'inversify'
-import { Collection, Db } from 'mongodb'
 import * as Pino from 'pino'
 import { pick, pipeP } from 'ramda'
 
 import { childWithFileName } from 'Helpers/Logging'
 
+import { BlockchainInfo, WalletInfo, NetworkInfo, IPFSInfo } from './HealthDAO'
+
```
Comment (lautarodragan) [inline src/Health/HealthController.ts]: Can you merge these two import lines?

**Comment by wzalazar** [comment]
LGTM, for me too

**Pull request merged**